openstack/networking/v2/subnets/requests.go

```diff
@@ -139,6 +140,8 @@ func (opts CreateOpts) ToSubnetCreateMap() (map[string]interface{}, error) {
 	}
 	if opts.GatewayIP != "" {
 		s["gateway_ip"] = opts.GatewayIP
+	} else if opts.NoGateway {
```

Comment:
It should be an error to provide `NoGateway` == ` true` and `GatewayIP` != "". We should probably check for that at the beginning of this `if` block. Also the one in `Update`.